boards/nxp/fmuk66-v3/src/spi.cpp

```diff
@@ -211,7 +211,7 @@ __EXPORT int fmuk66_spi_bus_initialize(void)
 
 	/* Configure SPI-based devices */
 
-	struct spi_dev_s *spi_sensors = px4_spibus_initialize(1);
+	struct spi_dev_s *spi_sensors = px4_spibus_initialize(PX4_BUS_NUMBER_TO_PX4(1));
```

Comment:
Do we actually need to map the bus number at all? In the board (i2c/spi) we're adding an offset (PX4_BUS_NUMBER_FROM_PX4), then in the micro_hal we're removing it (PX4_BUS_NUMBER_FROM_PX4).

Where would it break down if we simply used the real underlying bus number end-to-end? It's not really user facing (no external port numbering), and in the few cases it kind of is (decoding device ids?) it adds even more confusion.

What am I missing?

Reply:
The mapping was done to normalize to the STM32 way of counting busses 1-N.  We can change it but does it ripple to `-b` parameter somewhere? If it does common cod will not work.

Reply:
There isn't common code like that, it's all board specific and I would argue it makes the standalone `-b` usage far more confusing.

Reply:
For I2C it is used as an array index. I didn't want to go through that within these changes, so I left the offset in for now.
I'm all for removing it as well.